src/briefcase/console.py

```diff
@@ -34,6 +70,17 @@ def boolean_input(self, question, default=False):
         :param default: (optional) Default response (True/False)
         :returns: True if the user selected "y", or False if they selected "n".
         """
+
+        def normalize_input(user_input):
+            """Normalize user input to just the lowercase version of the first character."""
+            return user_input.lower()[:1]
+
+        def is_y_or_n(user_input):
+            """Validator for text_input() that user entered a literal 'y' or 'n'."""
+            if normalize_input(user_input) in {'y', 'n'}:
+                return True
+            raise ValueError("Please enter Y or N")
```

Comment:
This is... very complicated for a "is the lower case input y or n". 2 closures, one referencing the other... something that was, in the previous implementation, a 1 line lambda.

Reply:
Reverted the input handling refactor.